### PR TITLE
Add guide for Neptune usage

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What
Create a guide for Neptune usage in CMD, to aid the migration for developers. This will be used as a reference when running a Neptune setup session for developers, and can also be used in future as a reference.

### How to review
Check the changes make sense. Consider if anything is missing.

### Who can review
Anyone

### CODE REVIEW

File: guides/NEPTUNE.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+- Neptune supports both Gremlin and SPARQL as graph query languages. Our implementation is using Gremlin.
+- A Neptune cluster has a primary instance for read and write operations, and up to 15 read replica instances. When connecting to Neptune, the cluster endpoint will connect to the primary instance. There is also a reader endpoint which will connect to one of the available read only instances.
+- Neptune mostly complies with Gremlin standards but there are some caveats. This is the reason we cannot simply run another Gremlin compliant graph database locally. [See here](https://docs.aws.amazon.com/neptune/latest/userguide/feature-overview-standards-compliance.html) for more details on the standard compliance.
+- For more information on Neptune refer to the [Neptune docs](https://docs.aws.amazon.com/neptune/latest/userguide/intro.html)
+
+### Local development - How do I run the CMD services against Neptune?
+
+Neo4j runs within a docker container as part of the [dp-compose](https://github.com/ONSdigital/dp-compose) project. Running a local instance of Neptune is not possible, so we must use a test instance hosted in AWS.
```
Comment (DavidSubiros) [inline guides/NEPTUNE.md:16]: Just a comment for the future: 
There are some dockerised graph DBs that support gremlin. (e.g. https://hub.docker.com/r/tinkerpop/gremlin-server/) - would we consider adding something like this to dp-compose to have a fully local environment in the future? Or maybe it's not worth it?

File: guides/NEPTUNE.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    
+    ![1](../img/neptune_cluster_address.png)
+
+1. In the terminal, CD into the ansible directory of the dp-setup project:
+```
+cd dp-setup/ansible
+```
+1. Run the command to port forward to the Neptune cluster:
+```
+ssh -F ssh.cfg -L 8182:{cluster address}:8182 {user name}@{publishing asg node IP}
```
Comment (DavidSubiros) [inline guides/NEPTUNE.md:38]: we may want to add the dp tool alternative, to do ssh? Would it work?
```
dp ssh -p 8182:{cluster address}:22521 develop publishing {num}
```
where num corresponds to the node number with the as node IP that you want

**Review by DavidSubiros — APPROVED** [review]
Approved, with a couple of comments maybe for the future?

File: guides/NEPTUNE.md (hunk 1 above)
Comment (CarlHembrough) [reply guides/NEPTUNE.md:16]: It was attempted during the early phases of the Neptune migration but unfortunately did not work. This is due to Neptune not being fully compliant with the Gremlin standards. I touch on this on line 11, but maybe I can add some more to emphasise it?

File: guides/NEPTUNE.md (hunk 2 above)
Comment (CarlHembrough) [reply guides/NEPTUNE.md:38]: Good point. I had it on my todo to investigate a custom dp tool action to connect to Neptune, however this looks like a better solution for the short term. I will give it a go 👍

**Pull request merged**